source/developers-guide/concept-cart-bundle/index.md

```diff
@@ -0,0 +1,424 @@
+    }
+
+    /**
+     * Access point for cart process
+     * @param Cart $cart
+     * @param ProcessorCart $processorCart
+     * @param CartContextInterface $context
+     */
+    public function process(
+        Cart $cart,
+        ProcessorCart $processorCart,
+        CartContextInterface $context
+    ) {
+        //no logged in state?
+        if (!$context->getCustomer()) {
+            return;
+        }
+
+        //validate if customer should get discount
+        if (!$this->isNewCustomer($context->getCustomer()->getId())) {
+            return;
+        }
+
+        //get access to all goods inside the cart
+        $goods = $processorCart->getLineItems()->filterGoods();
+
+        //use core calculator for percentage price calculation with all goods prices
+        $discount = $this->percentagePriceCalculator->calculatePrice(
+            -10,
+            $goods->getPrices(),
+            $context
+        );
+
+        //add calculated discount to cart
+        $processorCart->getLineItems()->add(
+            new CalculatedDiscount('new-customer-discount', $discount)
+        );
+    }
+
+    /**
+     * Validates if the provided customer id is a new customer in the system
+     * and should get the `new customer discount`
+     *
+     * @param int $customerId
+     * @return bool
+     */
+    private function isNewCustomer($customerId)
+    {
+        $hasOrder = $this->connection->createQueryBuilder()
+            ->select('1')
+            ->from('s_order', 'orders')
+            ->where('orders.userID = :id')
+            ->andWhere('orders.ordernumber > 0')
+            ->setMaxResults(1)
+            ->setParameter(':id', $customerId)
+            ->execute()
+            ->fetch(\PDO::FETCH_COLUMN);
+
+        return !$hasOrder;
+    }
+}
+```
+First two conditions validates if a customer is logged in and if the customer has already taken an order.
+After the validation passes that the customer is a new customer, the processor collects first all calculated goods in the cart `$goods = $processorCart->getLineItems()->filterGoods();`.
+To calculate the percentage discount for the `new customer discount` the processor uses the shopware core calculator `\Shopware\Bundle\CartBundle\Domain\Price\PercentagePriceCalculator`.
+
+The processor has to be registered over the `cart_processor` container tag. The priority defines at which position the calculator has to be executed (after product calculation, before voucher, ...).
+```
+<service id="cart_extension.new_customer_discount_processor" class="CartExtension\Bundle\CartBundle\Domain\NewCustomerDiscountProcessor">
+    <argument type="service" id="dbal_connection" />
+    <argument type="service" id="shopware_cart.percentage_price_calculator" />
+    <tag name="cart_processor" priority="20000" />
+</service>
+```
+
+### Example - Blacklisted products
+The following examples shows, a possible solution, to prevent that some products can be inside the cart:
```

Comment:
The following examples shows a possible solution for preventing some products from entering the cart: